src/main.ts

```diff
@@ -18,61 +64,64 @@ interface IManifestRepoData extends IBasicRepoData {
     path: string;
 }
 
+interface IIssueContent {
+    title: string;
+    body: string;
+    labels: string[];
+}
 
-dotenv.config();
+export enum ToolName {
+    Node = "Node",
+    Python = "Python",
+    Go = "Go",
+}
 
-function compareDates(date: string): boolean {
-    try {
-        const currentDate = new Date().toISOString().split('T')[0];
-        return date >= currentDate;
-    } catch (error) {
-        core.setFailed((error as Error).message);
-        return false;
-    }
+function compareDateToCurrent(date: string): boolean {
+    const currentDate = new Date().toISOString().split('T')[0];
+    return date >= currentDate;
+}
+
+// Function used for Go versions only
+function calculateSixMonthsFromGivenDate(givenDate: Date): string {
+    const sixMonthsFromGivenDate = new Date(givenDate.setMonth(givenDate.getMonth() + 6));
+    return sixMonthsFromGivenDate.toISOString().split('T')[0];
 }
 
 function isMoreThanSixMonthsApart(givenDate: Date): boolean {
-    try {
-        const currentDate = new Date();
-        const timeDifferenceMs = givenDate.getTime() - currentDate.getTime();
-        const monthsDifference = Math.floor(timeDifferenceMs / (1000 * 60 * 60 * 24 * 30));
-        const yearsDifference = Math.floor(monthsDifference / 12);    
-        return yearsDifference > 0 || monthsDifference > 6;
-    } catch (error) {
-        core.setFailed((error as Error).message);
-        return false;
-    }
+    const currentDate = new Date();
+    const timeDifferenceMs = givenDate.getTime() - currentDate.getTime();
+    const monthsDifference = Math.floor(timeDifferenceMs / (1000 * 60 * 60 * 24 * 30));
```

Comment:
Just for readability
```suggestion
    const msPerMonth = 1000 * 60 * 60 * 24 * 30;
    const monthsDifference = Math.floor(timeDifferenceMs / msPerMonth);
```